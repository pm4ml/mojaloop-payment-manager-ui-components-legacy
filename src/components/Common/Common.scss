@import '../../assets/styles/vars/colors.scss';
@import '../../assets/styles/vars/fonts.scss';

.validation__icon {
  fill: transparent;
}

.validation__icon.validation__icon--invalid {
  height: auto;
  width: auto;
  fill: $color__main-danger;
}

.validation__messages {
  list-style-type: none;
  margin: 0;
  padding: 14px;
  padding-right: 18px;
  background: $color__gray-1;
  box-shadow: 0px 1px 2px $color__alpha-2;
  border-radius: 3px;
  min-width: 60px;
  display: block;
}

.validation__message {
  font-family: $font__primary, $font__primary-fallback;
  letter-spacing: $font__primary-spacing;
  display: flex;
  align-items: center;
  font-size: 12px;
  color: $color__dark-gray-1;
  min-width: 60px;
}

.validation__message--inactive {
  text-decoration: line-through;
  stroke: none;
  color: $color__gray-5;
}

.validation__message__text {
  white-space: nowrap;
}

.validation__message__icon-container {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 16px;
  height: 16px;
  margin-right: 8px;
}

.validation__message__icon--active {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-danger;
  stroke: $color__main-danger;
}

.validation__message__icon--inactive {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  fill: $color__main-success;
}

.validation__message__icon--unset {
  animation: fadeInValidationMessageIcon 0.2s ease-out;
  height: 6px;
  width: 6px;
  margin: 3px;
  border-radius: 3px;
  display: flex;
  background: #ddd;
}

@keyframes fadeInValidationMessageIcon {
  0% {
    transform: scale(0.5, 0.5);
    opacity: 0.5;
  }

  100% {
    transform: scale(1, 1);
    opacity: 1;
  }
}

.named__icon {
  height: 20px;
  width: 20px;
  border-radius: 20px;
  font-size: 13px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: #022843;
  color: #fff;
  text-transform: uppercase;
  text-decoration: none;

  &:hover {
    text-decoration: none;
  }
}
