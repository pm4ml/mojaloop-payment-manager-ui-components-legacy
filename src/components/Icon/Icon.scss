@import '../../assets/styles/vars/colors.scss';

.el-icon {
  display: flex;
  flex: 0 0 auto;
}

.el-icon--spin {
  animation: spinIcon 1s linear infinite;
}

@keyframes spinIcon {
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
}
