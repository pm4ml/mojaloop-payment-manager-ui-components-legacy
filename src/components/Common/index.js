import './Common.scss';
import '../../icons/mule/check-small.svg';
import '../../icons/mule/close-small.svg';

import PropTypes from 'prop-types';
import React from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon, { iconSizes } from '../Icon';
import Row from '../Row';
import Spinner from '../Spinner';
import Tooltip from '../Tooltip';

const Loader = ({ size }) => (
  <div className="mb-input__inner-icon mb-loader">
    <Spinner size={iconSizes[size]} />
  </div>
);

Loader.propTypes = {
  size: PropTypes.oneOf(['s', 'm', 'l']),
};
Loader.defaultProps = {
  size: 'l',
};

const Placeholder = ({ label, size, active }) => {
  // The Placeholder that renders inside an input

  const placeholderClassName = utils.composeClassNames([
    'mb-input__placeholder',
    size === 's' && 'mb-input__placeholder--small',
    size === 'm' && 'mb-input__placeholder--medium',
    size === 'l' && 'mb-input__placeholder--large',
    size === 's' && active && 'mb-input__placeholder--active-small',
    size === 'm' && active && 'mb-input__placeholder--active-medium',
    size === 'l' && active && 'mb-input__placeholder--active-large',
  ]);

  return <label className={placeholderClassName}>{label}</label>;
};

Placeholder.propTypes = {
  label: PropTypes.string,
  active: PropTypes.bool,
  size: PropTypes.oneOf(['s', 'm', 'l']),
};

Placeholder.defaultProps = {
  label: undefined,
  active: false,
  size: 'l',
};

const InnerButton = ({ className, size, kind, onClick, label, disabled, noFill, icon }) => {
  // Internal button used by inputs

  let innerButtonSize;
  if (size === 'l') {
    innerButtonSize = 'm';
  } else if (size === 'm') {
    innerButtonSize = 's';
  } else {
    innerButtonSize = 'xs';
  }

  const innerButtonClassName = utils.composeClassNames([
    className,
    'mb-input__inner-button',
    size === 's' && 'mb-input__inner-button--small',
    size === 'm' && 'mb-input__inner-button--medium',
    size === 'l' && 'mb-input__inner-button--large',
  ]);
  return (
    <Button
      kind={kind}
      size={innerButtonSize}
      className={innerButtonClassName}
      icon={icon}
      noFill={noFill}
      onClick={onClick}
      tabIndex="-1"
      label={label}
      disabled={disabled}
    />
  );
};

InnerButton.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf(['s', 'm', 'l']),
  kind: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  label: PropTypes.string,
  disabled: PropTypes.bool,
};

InnerButton.defaultProps = {
  className: undefined,
  size: 'l',
  kind: 'primary',
  icon: undefined,
  onClick: undefined,
  label: undefined,
  disabled: false,
};

const ActiveValidationIcon = () => (
  <Icon className="validation__message__icon--active" name="close-small" size={12} />
);
const InactiveValidationIcon = () => (
  <Icon className="validation__message__icon--inactive" name="check-small" size={14} />
);
const UnsetValidationIcon = () => <div className="validation__message__icon--unset" />;

const ValidationIcon = ({ active }) => {
  if (active === true) {
    return <ActiveValidationIcon />;
  } else if (active === false) {
    return <InactiveValidationIcon />;
  }
  return <UnsetValidationIcon />;
};

const ValidationMessage = ({ message, active }) => (
  <Row>
    <li
      className={`validation__message ${active === false ? 'validation__message--inactive' : ''}`}
    >
      <div className="validation__message__icon-container">
        <ValidationIcon active={active} />
      </div>
      <span className="validation__message__text">{message}</span>
    </li>
  </Row>
);

ValidationMessage.propTypes = {
  message: PropTypes.string,
  active: PropTypes.bool,
};
ValidationMessage.defaultProps = {
  message: undefined,
  active: undefined,
};

const ValidationMessages = ({ messages }) => {
  let validationMessageList = null;
  if (messages.length) {
    validationMessageList = messages.map(({ message, active }, i) => (
      <ValidationMessage key={i.toString()} message={message} active={active} />
    ));
  }
  return <ul className="validation__messages">{validationMessageList}</ul>;
};

ValidationMessages.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape(ValidationMessage.propTypes)),
};

ValidationMessages.defaultProps = {
  messages: [],
};

const InvalidIcon = ({ size }) => (
  <Icon
    size={iconSizes[size]}
    name="warning-sign"
    className="mb-input__inner-icon validation__icon validation__icon--invalid "
    tooltip="Invalid"
    tooltipDelay={500}
    tooltipKind="error"
  />
);

const ValidationWrapper = ({ messages, active, children }) => (
  <Tooltip
    position="right"
    kind="neutral"
    custom
    delay={500}
    content={<ValidationMessages messages={messages} />}
    forceVisibility={active === true && messages && messages.length > 0}
    showOnHover={false}
  >
    {children}
  </Tooltip>
);

const NamedIcon = ({ namedIconTitle }) => <div className="named__icon">{namedIconTitle[0]}</div>;

NamedIcon.propTypes = {
  namedIconTitle: PropTypes.string,
};
NamedIcon.defaultProps = {
  namedIconTitle: undefined,
};

export { Loader, Placeholder, InnerButton, ValidationWrapper, InvalidIcon, NamedIcon };
