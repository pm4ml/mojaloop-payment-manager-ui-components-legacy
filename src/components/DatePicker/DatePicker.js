import '../../icons/mule/calendar-small.svg';
import './DatePicker.scss';

import moment from 'moment';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import DayPicker, { DateUtils } from 'react-day-picker';

import * as utils from '../../utils/common';
import keyCodes from '../../utils/keyCodes';
import { InvalidIcon, Loader, Placeholder, ValidationWrapper } from '../Common';
import Icon, { iconSizes } from '../Icon';

function asDate(day) {
  if (day) {
    return moment(day)
      .startOf('day')
      .format('x');
  }
  return 0;
}

class DatePicker extends PureComponent {
  static getDate(value) {
    if (typeof value === 'string' || typeof value === 'number') {
      return new Date(parseInt(value, 10));
    }
    return undefined;
  }
  static getTimeObject(value) {
    let intValue = value;
    if (typeof value === 'undefined') {
      intValue = 0;
    }
    if (typeof value === 'string') {
      intValue = parseInt(value, 10);
    }

    const hour = value ? moment(intValue).get('hour') : 0;
    const minute = value ? moment(intValue).get('minute') : 0;
    const second = value ? moment(intValue).get('second') : 0;
    return { hour, minute, second };
  }
  static getSelectedDayAndTimestamp(value) {
    let myValue = value;
    if (typeof value === 'string') {
      myValue = parseInt(value, 10);
    }
    return {
      selectedDay: myValue ? new Date(myValue) : null,
      timestamp: myValue,
    };
  }
  static getTimestamp(day, hour, minute, second) {
    const date = asDate(day);
    return parseInt(date, 10) + hour * 60 * 60 * 1000 + minute * 60 * 1000 + second * 1000;
  }

  constructor(props) {
    super(props);

    this.handleDateTimeChange = this.handleDateTimeChange.bind(this);
    this.handleDayClick = this.handleDayClick.bind(this);
    this.handleHourClick = this.handleHourClick.bind(this);
    this.handleMinuteClick = this.handleMinuteClick.bind(this);
    this.handlePageClick = this.handlePageClick.bind(this);
    this.handleResize = this.handleResize.bind(this);
    this.handleSecondClick = this.handleSecondClick.bind(this);

    // Wrapper events
    this.onClick = this.onClick.bind(this);
    this.onBlur = this.onBlur.bind(this);
    this.onFocus = this.onFocus.bind(this);

    // internal methods
    this.closeDatePicker = this.closeDatePicker.bind(this);
    this.leaveDatePicker = this.leaveDatePicker.bind(this);
    this.testKey = this.testKey.bind(this);

    // internal timeout
    this.deferredTimeChangeTimeout = undefined;

    const { selectedDay, timestamp } = DatePicker.getSelectedDayAndTimestamp(this.props.value);
    const { hour, minute, second } = DatePicker.getTimeObject(this.props.value);

    this.state = {
      selectedDay,
      timestamp,
      hour,
      minute,
      second,
      isOpen: false,
    };
  }

  componentDidMount() {
    window.addEventListener('resize', this.handleResize);
    window.addEventListener('mouseup', this.handlePageClick, false);
  }
  componentDidUpdate(prevProps) {
    const { value, defaultHour, defaultMinute, defaultSecond } = this.props;
    const defaultTime = {
      hour: defaultHour,
      minute: defaultMinute,
      second: defaultSecond,
    };

    if (prevProps.value !== value) {
      let timeValue = value;
      if (typeof value === 'string') {
        timeValue = parseInt(value, 10);
      }
      const { selectedDay, timestamp } = DatePicker.getSelectedDayAndTimestamp(timeValue);
      const { hour, minute, second } = value ? DatePicker.getTimeObject(timeValue) : defaultTime;

      this.setState({
        selectedDay,
        timestamp,
        hour,
        minute,
        second,
      });
    }
  }
  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
    window.removeEventListener('mouseup', this.handlePageClick, false);
  }
  onClick(e) {
    if (this.props.onClick) {
      this.props.onClick(e);
    }
    this.input.focus();
  }
  onFocus(e) {
    if (this.state.isOpen === false) {
      this.setState({ isOpen: true }, () => {
        this.input.focus();
      });
      this.handleResize();
    }
    if (this.props.onFocus) {
      this.props.onFocus(e);
    }
  }
  onBlur(e) {
    if (this.props.onBlur) {
      this.props.onBlur(e);
    }
    // this.close();
  }
  closeDatePicker(e) {
    this.setState({ isOpen: false });
    this.onBlur(e);
  }
  handleDateTimeChange(selectedDay, hour, minute, second) {
    const { format } = this.props;
    // convert the value into the specified format if necessary
    let exportDay = selectedDay === null || selectedDay === undefined ? undefined : selectedDay;

    if (exportDay !== undefined && format) {
      const dayStamp = moment(exportDay)
        .startOf('day')
        .format('x');

      exportDay =
        parseInt(dayStamp, 10) + hour * 60 * 60 * 1000 + minute * 60 * 1000 + second * 1000;

      if (format === 'x') {
        exportDay = parseInt(exportDay, 10);
      } else if (format === 'X') {
        exportDay = parseInt(exportDay / 1000, 10);
      } else {
        exportDay = moment(exportDay).format(format);
      }
    }

    // call the external function
    if (this.props.onSelect) {
      this.props.onSelect(exportDay);
    }
  }
  handleDayClick(day, { selected, disabled }) {
    if (disabled) return;
    const selectedDay = selected ? null : day;
    const timestamp = DatePicker.getTimestamp(selected ? undefined : day, 0, 0, 0);
    this.setState({ selectedDay, timestamp });
    this.handleDateTimeChange(selectedDay, this.state.hour, this.state.minute, this.state.second);
  }
  handleHourClick(hour) {
    clearTimeout(this.deferredTimeChangeTimeout);
    const timestamp = DatePicker.getTimestamp(
      this.state.selectedDay,
      hour,
      this.state.minute,
      this.state.second,
    );
    if (hour > 23) return;
    this.setState({ hour, timestamp });
    this.deferredTimeChangeTimeout = setTimeout(
      () =>
        this.handleDateTimeChange(
          this.state.selectedDay,
          hour,
          this.state.minute,
          this.state.second,
        ),
      500,
    );
  }
  handleMinuteClick(minute) {
    clearTimeout(this.deferredTimeChangeTimeout);
    const timestamp = DatePicker.getTimestamp(
      this.state.selectedDay,
      this.state.hour,
      minute,
      this.state.second,
    );
    if (minute > 59) return;
    this.setState({ minute, timestamp });
    this.deferredTimeChangeTimeout = setTimeout(
      () =>
        this.handleDateTimeChange(
          this.state.selectedDay,
          this.state.hour,
          minute,
          this.state.second,
        ),
      500,
    );
  }
  handlePageClick(e) {
    if (!this.calendarPosition.contains(e.target)) {
      this.setState({ isOpen: false });
    }
  }
  handleResize() {
    const wrapper = utils.getParentOverflow(this.calendarPosition);
    const calendarHeight = 300;
    const { maxLowerHeight, maxUpperHeight } = utils.getSpaceAvailability(
      calendarHeight,
      this.calendarPosition,
      wrapper,
    );
    this.reverse = maxLowerHeight > calendarHeight ? false : maxLowerHeight < maxUpperHeight;

    clearTimeout(this._forceUpdateTimeout);
    this._forceUpdateTimeout = setTimeout(() => this.forceUpdate(), 50);
  }
  handleSecondClick(second) {
    clearTimeout(this.deferredTimeChangeTimeout);
    const timestamp = DatePicker.getTimestamp(
      this.state.selectedDay,
      this.state.hour,
      this.state.minute,
      second,
    );
    if (second > 59) return;
    this.setState({ second, timestamp });
    this.deferredTimeChangeTimeout = setTimeout(
      () =>
        this.handleDateTimeChange(
          this.state.selectedDay,
          this.state.hour,
          this.state.minute,
          second,
        ),
      500,
    );
  }
  leaveDatePicker(e, next) {
    this.closeDatePicker(e);
    utils.focusNextFocusableElement(this.input, next);
  }
  testKey(e) {
    const { keyCode, shiftKey } = e.nativeEvent;
    if (keyCode === keyCodes.KEY_TAB) {
      e.preventDefault();
      this.leaveDatePicker(e, !shiftKey);
    }
  }

  render() {
    const {
      placeholder,
      id,
      className,
      size,
      style,
      disabled,
      pending,
      invalid,
      invalidMessages,
      required,
      dateFormat,
      hideIcon,
    } = this.props;
    const { isOpen, timestamp, selectedDay } = this.state;
    const initialMonth = selectedDay || DatePicker.getDate(this.props.initialMonth);
    const hasDate = timestamp !== 0 && timestamp !== undefined;
    const isPlaceholderActive = isOpen || hasDate;
    const iconSize = iconSizes[size];
    const valueFormat = dateFormat || 'MMM Do YYYY, HH:mm:ss';
    const value = hasDate ? moment(timestamp).format(valueFormat) : '';
    const showCalendar = hasDate ? hideIcon === false : true;

    const componentClassName = utils.composeClassNames([
      className,
      'input-datepicker__component',
      'mb-input',
      'mb-input__borders',
      'mb-input__background',
      'mb-input__shadow',
      size === 's' && 'mb-input--small',
      size === 'm' && 'mb-input--medium',
      size === 'l' && 'mb-input--large',
      /* eslint-disable max-len  */
      isOpen &&
        'mb-input--open mb-input__borders--open mb-input__background--open mb-input__shadow--open',
      disabled && 'mb-input--disabled mb-input__borders--disabled mb-input__background--disabled',
      pending &&
        'mb-input--pending mb-input__borders--pending mb-input__background--pending mb-input__shadow--pending',
      invalid &&
        'mb-input--invalid mb-input__borders--invalid mb-input__background--invalid mb-input__shadow--invalid',
      required &&
        'mb-input--required mb-input__borders--required mb-input__background--required mb-input__shadow--required',
      /* eslint-enable */
    ]);

    let customPlaceholder = null;
    if (placeholder) {
      customPlaceholder = (
        <Placeholder size={size} label={placeholder} active={isPlaceholderActive} />
      );
    }

    let loader = null;
    if (pending) {
      loader = <Loader size={size} />;
    }

    let invalidIcon = null;
    if (invalid) {
      invalidIcon = <InvalidIcon size={size} />;
    }

    let calendarIcon = null;
    if (showCalendar) {
      calendarIcon = (
        <div className="mb-input__inner-icon input-datepicker__icon">
          <Icon size={iconSize} name="calendar-small" fill="#999" />
        </div>
      );
    }

    const calendarBoxClassName = utils.composeClassNames([
      'input-datepicker__calendar-box',
      this.reverse && 'input-datepicker__calendar-box--reverse',
    ]);

    return (
      <ValidationWrapper messages={invalidMessages} active={isOpen} key="datepicker">
        <div id={id} className={componentClassName} style={style}>
          <div
            className="mb-input__content input-datepicker__content"
            onClick={this.onClick}
            onKeyDown={this.onFocus}
            role="presentation"
          >
            {customPlaceholder}
            <input
              onFocus={this.onFocus}
              ref={input => {
                this.input = input;
              }}
              className="mb-input__input input-datepicker__value"
              value={value}
              onKeyDown={this.testKey}
              disabled={disabled}
              readOnly
            />
            {loader}
            {invalidIcon}
            {calendarIcon}
          </div>
        </div>
        <div
          key="calendar"
          className="input-datepicker--position"
          ref={calendarPosition => {
            this.calendarPosition = calendarPosition;
          }}
        >
          {this.state.isOpen && (
            <div className={calendarBoxClassName}>
              <DayPicker
                initialMonth={initialMonth}
                selectedDays={day => DateUtils.isSameDay(selectedDay, day)}
                onDayClick={this.handleDayClick}
                disabledDays={this.props.disabledDays}
              />
              {this.props.withTime && (
                <TimePicker
                  hour={this.state.hour}
                  minute={this.state.minute}
                  second={this.state.second}
                  onHourChange={this.handleHourClick}
                  onMinuteChange={this.handleMinuteClick}
                  onSecondChange={this.handleSecondClick}
                  disabled={selectedDay === undefined || selectedDay == null}
                />
              )}
            </div>
          )}
        </div>
      </ValidationWrapper>
    );
  }
}

DatePicker.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  style: PropTypes.shape(),
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  placeholder: PropTypes.string,
  size: PropTypes.oneOf(['s', 'm', 'l']),
  onSelect: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  format: PropTypes.string,
  dateFormat: PropTypes.string,
  withTime: PropTypes.bool,
  defaultHour: PropTypes.number,
  defaultMinute: PropTypes.number,
  defaultSecond: PropTypes.number,
  disabled: PropTypes.bool,
  pending: PropTypes.bool,
  required: PropTypes.bool,
  invalid: PropTypes.bool,
  initialMonth: PropTypes.string,
  disabledDays: PropTypes.func,
  hideIcon: PropTypes.bool,
};
DatePicker.defaultProps = {
  id: undefined,
  className: undefined,
  style: undefined,
  size: 'l',
  value: undefined,
  placeholder: undefined,
  onSelect: undefined,
  onBlur: undefined,
  onFocus: undefined,
  format: undefined,
  dateFormat: undefined,
  withTime: false,
  defaultHour: 0,
  defaultMinute: 0,
  defaultSecond: 0,
  disabled: false,
  pending: false,
  required: false,
  invalid: false,
  initialMonth: undefined,
  disabledDays: undefined,
  hideIcon: false,
};

const TimePicker = ({
  hour,
  minute,
  second,
  onHourChange,
  onMinuteChange,
  onSecondChange,
  disabled,
}) => (
  <div className="timepicker-position">
    <TimeInput
      name="Hours"
      limit={23}
      selected={hour}
      onChange={onHourChange}
      disabled={disabled}
    />
    <TimeInput
      name="Minutes"
      limit={59}
      selected={minute}
      onChange={onMinuteChange}
      disabled={disabled}
    />
    <TimeInput
      name="Seconds"
      limit={59}
      selected={second}
      onChange={onSecondChange}
      disabled={disabled}
    />
  </div>
);
TimePicker.propTypes = {
  hour: PropTypes.number,
  minute: PropTypes.number,
  second: PropTypes.number,
  onHourChange: PropTypes.func,
  onMinuteChange: PropTypes.func,
  onSecondChange: PropTypes.func,
  disabled: PropTypes.bool,
};
TimePicker.defaultProps = {
  hour: 0,
  minute: 0,
  second: 0,
  onHourChange: undefined,
  onMinuteChange: undefined,
  onSecondChange: undefined,
  disabled: false,
};

class TimeInput extends PureComponent {
  static getDoubleDigitTime(value) {
    const intValue = parseInt(value, 10);
    return intValue < 10 ? `0${intValue}` : intValue;
  }
  constructor(props) {
    super(props);

    this.onChangeValue = this.onChangeValue.bind(this);
    this.state = {
      value: parseInt(this.props.selected, 10),
      shown: TimeInput.getDoubleDigitTime(this.props.selected),
    };
  }

  componentDidUpdate(prevProps) {
    const { selected } = this.props;
    if (prevProps.selected !== this.props.selected) {
      if (parseInt(selected, 10) === this.state.value) return;
      this.setState({
        value: parseInt(selected, 10),
        shown: TimeInput.getDoubleDigitTime(selected),
      });
    }
  }

  onChangeValue(e) {
    const { value } = e.target;
    const time = value === '' ? 0 : parseInt(value, 10);
    this.props.onChange(time);
  }

  render() {
    const { name, limit, disabled } = this.props;
    return (
      <div className="timepicker-item-box">
        <div className="placeholder"> {name} </div>
        <input
          disabled={disabled}
          type="number"
          className="timepicker-select"
          onFocus={e => e.target.select()}
          onChange={this.onChangeValue}
          value={this.state.shown}
          min={0}
          max={limit}
        />
      </div>
    );
  }
}
TimeInput.propTypes = {
  name: PropTypes.string,
  limit: PropTypes.number,
  selected: PropTypes.number,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
};
TimeInput.defaultProps = {
  name: '',
  limit: 0,
  selected: false,
  onChange: undefined,
  disabled: false,
};
export default DatePicker;
