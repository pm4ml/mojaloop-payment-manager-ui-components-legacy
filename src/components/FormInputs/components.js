import './FormInput.scss';
import '../../icons/mule/forum-small.svg';

import React, { PureComponent } from 'react';

import * as utils from '../../utils/common';
import Button from '../Button';
import Icon from '../Icon';
import Tooltip from '../Tooltip';

export const FieldInfoOverlay = ({ assignRef, title, description, url, required }) => (
  <div className="forminput__field-info" ref={assignRef}>
    <div className="forminput__field-info__title-container">
      <div className="forminput__field-info__title">{title}</div>
      {required && (
        <React.Fragment>
          <div className="forminput__field-info__required-icon">
            <Icon size={20} name="info-small" className="forminput__label-icon" fill="#333" />
          </div>
          <div className="forminput__field-info__required-label"> required </div>
        </React.Fragment>
      )}
    </div>
    <div className="forminput__field-info__content">
      <p>{description}</p>
    </div>
    {url && (
      <div className="forminput__field-info__url">
        <a href={url}>{url}</a>
      </div>
    )}
  </div>
);

export class FieldInfo extends PureComponent {
  constructor() {
    super();
    this.onClick = this.onClick.bind(this);
    this.open = this.open.bind(this);
    this.close = this.close.bind(this);
    this.closeIfClickingOutside = this.closeIfClickingOutside.bind(this);
    this.state = {
      open: false,
    };
  }
  componentWillUnmount() {
    this.close();
  }
  onClick() {
    if (this.state.open) {
      this.close();
    } else {
      this.open();
    }
  }
  open() {
    this.area = React.createRef();
    this.setState(
      {
        open: true,
      },
      () => {
        document.addEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  close() {
    this.setState(
      {
        open: false,
      },
      () => {
        this.area = null;
        document.removeEventListener('click', this.closeIfClickingOutside);
      },
    );
  }
  closeIfClickingOutside(evt) {
    if (!this.area.current) {
      return;
    }
    if (!this.area.current.contains(evt.target)) {
      this.close();
    }
  }
  render() {
    const { iconSize, title, required, description, url } = this.props;
    let content = null;
    if (this.state.open) {
      content = (
        <FieldInfoOverlay
          assignRef={this.area}
          title={title}
          description={description}
          url={url}
          required={required}
        />
      );
    }
    const fieldInfoClassName = utils.composeClassNames([
      'forminput__field-info__icon',
      this.state.open && 'forminput__field-info__icon--active',
    ]);

    return (
      <Tooltip
        forceVisibility={this.state.open}
        showOnHover={false}
        content={content}
        custom
        position="right"
        align="start"
      >
        <Icon
          name="forum-small"
          onClick={this.onClick}
          size={iconSize}
          className={fieldInfoClassName}
        />
      </Tooltip>
    );
  }
}

export const Label = ({ size, label, required, complete, description, url }) => {
  if (!label) {
    return null;
  }

  const className = utils.composeClassNames([
    'forminput__label-box',
    size === 's' && 'forminput__label-box--small',
    size === 'm' && 'forminput__label-box--medium',
    size === 'l' && 'forminput__label-box--large',
  ]);
  const iconSizes = {
    s: 11,
    m: 12,
    l: 14,
  };
  const iconSize = iconSizes[size];
  return (
    <div className={className}>
      {required && (
        <Icon
          size={iconSize}
          name="info-small"
          className="forminput__label-icon"
          fill={complete ? '#39f' : '#f93'}
          tooltip={complete ? '' : 'This is a required field'}
        />
      )}
      <label>{label}</label>
      {description && (
        <FieldInfo
          iconSize={iconSize}
          title={label}
          description={description}
          url={url}
          required={required}
        />
      )}
    </div>
  );
};

export const InlineButton = ({ visible, isLocked, isDisabled, onClick, label }) => {
  if (!visible || isLocked || isDisabled) {
    return null;
  }

  return (
    <Button
      size="m"
      noFill
      kind="secondary"
      className="forminput__inline-button"
      onClick={onClick}
      disabled={isDisabled}
      label={label}
    />
  );
};

export const LockedIcon = ({ locked }) => {
  if (!locked) {
    return null;
  }
  return (
    <div className="forminput__inline-icon">
      <Icon
        name="lock-small"
        size={20}
        fill="#999"
        style={{ marginLeft: '10px' }}
        tooltip="this field is locked"
      />
    </div>
  );
};

export const InfoMessage = ({ message }) => (
  <div className="forminput__input-message">{message && <Tooltip>{message}</Tooltip>}</div>
);
