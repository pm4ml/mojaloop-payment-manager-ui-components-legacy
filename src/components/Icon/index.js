import Icon, { iconSizes } from './Icon';

export default Icon;
export { iconSizes };
