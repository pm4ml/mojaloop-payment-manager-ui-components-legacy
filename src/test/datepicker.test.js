import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import { InvalidIcon, Loader, Placeholder, ValidationWrapper } from '../components/Common';
import DatePicker from '../components/DatePicker';
import Icon from '../components/Icon';

it('renders the datepicker', () => {
  const wrapper = mount(<DatePicker />);
  expect(wrapper.find('div.input-datepicker__component')).toHaveLength(1);
});

it('renders the datepicker value', () => {
  const wrapper = mount(<DatePicker value="1526478000000" />);
  expect(wrapper.find('input').prop('value')).not.toBe('');
});

it('renders the placeholder', () => {
  const wrapper = mount(<DatePicker placeholder="test-datepicker" />);
  expect(wrapper.find(Placeholder)).toHaveLength(1);
  expect(wrapper.find(Placeholder).prop('label')).toBe('test-datepicker');
});

it('renders the validation wrapper', () => {
  const wrapper = shallow(<DatePicker placeholder="test-Select" />);
  expect(wrapper.find(ValidationWrapper)).toHaveLength(1);
});

it('applies the prop className', () => {
  const wrapper = mount(<DatePicker className="test" />);
  expect(wrapper.find('div.input-datepicker__component').hasClass('test')).toBeTruthy();
});

it('renders the prop id', () => {
  const wrapper = mount(<DatePicker id="testDatePickerId" />);
  expect(wrapper.find('div.input-datepicker__component').find('#testDatePickerId')).toHaveLength(1);
});

it('renders the disabled state', () => {
  const wrapper = mount(<DatePicker disabled />);
  expect(wrapper.find('input')).toHaveLength(1);
  expect(wrapper.find('input').prop('disabled')).toBe(true);
});

it('renders the icon', () => {
  const wrapper = mount(<DatePicker />);
  expect(wrapper.find(Icon)).toHaveLength(1);
});

it('renders the pending state', () => {
  const wrapper = mount(<DatePicker pending />);
  expect(wrapper.find(Loader)).toHaveLength(1);
});

it('renders the invalid state', () => {
  const wrapper = mount(<DatePicker invalid />);
  expect(wrapper.find(InvalidIcon)).toHaveLength(1);
});

it('renders the required state', () => {
  const wrapper = shallow(<DatePicker required />);
  expect(wrapper.find('.mb-input--required')).toHaveLength(1);
});

it('renders the large, medium, small sizes', () => {
  const sizes = {
    large: 'l',
    medium: 'm',
    small: 's',
  };
  Object.entries(sizes).forEach(([name, size]) => {
    const wrapper = shallow(<DatePicker size={size} />);
    const className = `mb-input--${name}`;
    expect(wrapper.find('.mb-input').hasClass(className)).toBeTruthy();
  });
});

// Events

it('renders the calendar when focused', () => {
  const wrapper = mount(<DatePicker />);
  wrapper.find('input').simulate('focus');
  expect(wrapper.find('.input-datepicker--position')).toHaveLength(1);
});

it('does not close the calendar when selecting a day', () => {
  const wrapper = mount(<DatePicker />);
  wrapper.find('input').simulate('focus');
  expect(wrapper.find('.input-datepicker--position')).toHaveLength(1);
  wrapper
    .find('.DayPicker-Day')
    .at(10)
    .simulate('click');
  expect(wrapper.find('.input-datepicker--position')).toHaveLength(1);
  expect(wrapper.find('input').prop('value')).not.toBe('');
});

it('triggers onFocus when focused', () => {
  const mockEvent = jest.fn();
  const wrapper = mount(<DatePicker onFocus={mockEvent} />);
  expect(mockEvent).not.toHaveBeenCalled();
  wrapper.find('input').simulate('focus');
  expect(mockEvent).toHaveBeenCalled();
});

// Snapshot

it('renders the datepicker correctly when multiple props are set', () => {
  const wrapper = shallow(<DatePicker value="test-value" id="test-id" />);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});
