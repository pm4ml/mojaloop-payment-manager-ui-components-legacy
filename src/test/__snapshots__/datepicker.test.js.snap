// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the datepicker correctly when multiple props are set 1`] = `
<ValidationWrapper
  active={false}
  key="datepicker"
>
  <div
    className="input-datepicker__component mb-input mb-input__borders mb-input__background mb-input__shadow mb-input--large"
    id="test-id"
  >
    <div
      className="mb-input__content input-datepicker__content"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="presentation"
    >
      <input
        className="mb-input__input input-datepicker__value"
        disabled={false}
        onFocus={[Function]}
        onKeyDown={[Function]}
        readOnly={true}
        value="Invalid date"
      />
      <div
        className="mb-input__inner-icon input-datepicker__icon"
      >
        <Icon
          fill="#999"
          name="calendar-small"
          size={16}
          spin={false}
        />
      </div>
    </div>
  </div>
  <div
    className="input-datepicker--position"
    key="calendar"
  />
</ValidationWrapper>
`;
